api/auth/ldap/ldap.go

```diff
@@ -0,0 +1,83 @@
+package ldap
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/vault/api"
+)
+
+type LDAPAuth struct {
+	mountPath string
+	username  string
+	password  string
+}
+
+type Password struct {
+	// The password as a plaintext string value.
+	FromString string
+}
```

Comment:
With the Userpass and Approle auth modules, we provide the ability to pass in the password FromString, FromEnv, or FromFile, just to allow for different use cases (and to keep consistent). 

Would you be able to provide the same thing here, or is there a reason you only chose to use FromString?

Reply:
I think, at the time, my reasoning was likely centered around this being used in more interactive scenarios given the nature of LDAP authentication.

After reading this, It reminded me that I actually have some projects I plan to use this in where having `FromFile` and `FromEnv` would be necessary so I'm positive others out there would benefit from it. 

Got those two methods added in as well as their test coverage, let me know if there's anything else!